### PR TITLE
chore: replace reth-rpc-types by alloy-rpc-types

### PR DESCRIPTION
**Pull request by greged93**
Removes `reth-rpc-types` from `reth-node-builder`.

Linked to #11067.

### CODE REVIEW

File: crates/node/builder/Cargo.toml [hunk 1]
```diff
@@ -62,6 +61,7 @@ reth-transaction-pool.workspace = true
 ## ethereum
 alloy-network.workspace = true
 alloy-primitives.workspace = true
+alloy-rpc-types = { workspace = true, features = ["engine"] }
```
Comment (greged93) [inline crates/node/builder/Cargo.toml:64]: I could also use `alloy-rpc-types-eth`, `alloy-rpc-types-engine` and `alloy-serde` instead. Don't know how much granular you want to get.

Comment (mattsse) [reply crates/node/builder/Cargo.toml:64]: this is fine

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
sorry very unfortunate timing... we just merged a pr that introduced conflicts

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by greged93** [comment]
Will rebase this afternoon!

**Comment by mattsse** [comment]
did it real quick, the latest changes actually made this easier hehe

**Pull request merged**